BestAlignmentParams does not compile on macos due to clang issues

-----------------------------------------

If you have a question about the RDKit or aren't sure if what you're seeing is the right behavior please use the Discussions tab above (https://github.com/rdkit/rdkit/discussions) instead of posting it here.

Please help us help you: We normally need the information below in order to answer questions or understand bug reports. If you do not provide the information requested we may not be able to help you and will probably close the issue.

PLEASE DELETE THIS SECTION AFTER YOU HAVE READ IT

-----------------------------------------

**Describe the bug**
Trying to build Release_2025_09_4 on macos fails due to no matching constructor for BestAlignmentParams

**To Reproduce**
```
[ 76%] Building CXX object Code/GraphMol/MolAlign/Wrap/CMakeFiles/rdMolAlign.dir/rdMolAlign.cpp.o
    cd /Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/build/Code/GraphMol/MolAlign/Wrap && /Applications/Xcode_15.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DBOOST_ALL_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_NO_CXX98_FUNCTION_BASE -DBOOST_SERIALIZATION_DYN_LINK -DBOOST_SYSTEM_NO_DEPRECATED -DRDKIT_DYN_LINK -DRDK_64BIT_BUILD -DRDK_BUILD_COORDGEN_SUPPORT -DRDK_BUILD_DESCRIPTORS3D -DRDK_BUILD_MAEPARSER_SUPPORT -DRDK_BUILD_THREADSAFE_SSS -DRDK_BUILD_XYZ2MOL_SUPPORT -DRDK_BUILD_YAEHMOP_SUPPORT -DRDK_HAS_EIGEN3 -DRDK_TEST_MULTITHREADED -DRDK_USE_BOOST_IOSTREAMS -DRDK_USE_BOOST_SERIALIZATION -DRDK_USE_STRICT_ROTOR_DEFINITION -DRDK_USE_URF -D_HAS_AUTO_PTR_ETC=0 -DrdMolAlign_EXPORTS -I/Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/External -I/Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/External/RingFamilies/RingDecomposerLib/src/RingDecomposerLib -I/Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/Code -I/Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/External/boost-numeric-bindings -I/private/var/folders/sm/xrr7tmqj20s7hrsh1qhfl1d40000gn/T/pip-build-env-p94nde6g/overlay/lib/python3.9/site-packages/numpy/_core/include -isystem /Library/Frameworks/Python.framework/Versions/3.9/include/python3.9 -isystem /Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/External/Eigen/eigen -isystem /Users/runner/work/rdkit-pypi/rdkit-pypi/conan/direct_deploy/boost/include -isystem /Users/runner/.conan2/p/b/zlib266a94c6bf77e/p/include -Wno-implicit-function-declaration -DCATCH_CONFIG_NO_CPP17_UNCAUGHT_EXCEPTIONS -Wall -Wextra -Wno-deprecated -Wno-unused-function -fno-strict-aliasing -Wno-format -Wno-logical-op-parentheses -fPIC -stdlib=libc++ -O3 -DNDEBUG -std=gnu++20 -arch arm64 -isysroot /Applications/Xcode_15.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -mmacosx-version-min=11.0 -fPIC -Wno-deprecated-builtins -Wno-deprecated-non-prototype -Wno-unused-parameter -MD -MT Code/GraphMol/MolAlign/Wrap/CMakeFiles/rdMolAlign.dir/rdMolAlign.cpp.o -MF CMakeFiles/rdMolAlign.dir/rdMolAlign.cpp.o.d -o CMakeFiles/rdMolAlign.dir/rdMolAlign.cpp.o -c /Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/Code/GraphMol/MolAlign/Wrap/rdMolAlign.cpp
    /Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/Code/GraphMol/MolAlign/Wrap/rdMolAlign.cpp:37:9: error: no matching constructor for initialization of 'BestAlignmentParams'
          : BestAlignmentParams(maxMatches_, symmetrizeTerminalGroups_, ignoreHs_,
            ^                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/Code/GraphMol/MolAlign/AlignMolecules.h:39:30: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 6 were provided
    struct RDKIT_MOLALIGN_EXPORT BestAlignmentParams {
                                 ^
    /Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/Code/GraphMol/MolAlign/AlignMolecules.h:39:30: note: candidate constructor (the implicit move constructor) not viable: requires 1 argument, but 6 were provided
    /Users/runner/work/rdkit-pypi/rdkit-pypi/build/temp.macosx-11.0-arm64-cpython-39/rdkit/Code/GraphMol/MolAlign/AlignMolecules.h:39:30: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 6 were provided
    1 error generated.
    make[2]: *** [Code/GraphMol/MolAlign/Wrap/CMakeFiles/rdMolAlign.dir/rdMolAlign.cpp.o] Error 1
    make[1]: *** [Code/GraphMol/MolAlign/Wrap/CMakeFiles/rdMolAlign.dir/all] Error 2
    make: *** [all] Error 2
```

**Expected behavior**
Builds fine, like it does on linux (gcc)

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Configuration (please complete the following information):**
 - RDKit version: Release_2025_09_4
 - OS: macosx arm64 (github macos-14)
 - Python version (if relevant): n/a
 - Are you using conda? no
 - If you are using conda, which channel did you install the rdkit from? n/a
 - If you are not using conda: how did you install the RDKit? n/a

--- a/Code/GraphMol/MolAlign/Wrap/rdMolAlign.cpp
+++ b/Code/GraphMol/MolAlign/Wrap/rdMolAlign.cpp
@@ -34,9 +34,9 @@ struct pyBestAlignmentParams : public BestAlignmentParams {
   pyBestAlignmentParams(int maxMatches_, bool symmetrizeTerminalGroups_,
                         bool ignoreHs_, int numThreads_, python::object map_,
                         python::object weights_)
-      : BestAlignmentParams(maxMatches_, symmetrizeTerminalGroups_, ignoreHs_,
+      : BestAlignmentParams{maxMatches_, symmetrizeTerminalGroups_, ignoreHs_,
                             numThreads_, std::vector<MatchVectType>(),
-                            nullptr) {
+                            nullptr} {
     unsigned int nAtms = 0;
     if (map_ != python::object()) {
       map = translateAtomMapSeq(map_);

